### PR TITLE
Final Features for Ordering/Receiving

### PR DESCRIPTION
**Pull request by bwwaaa**
These are the final additions to the Ordering/Receiving feature. Adhering to clean architecture, many classes have been created such as OrderingInteractor, ReceivingController, OrderingInputBoundary, ReceivingResponseModel, and more that allow for proper independency following clean architecture principles. Along with the clean architecture classes, two additional tests were added: OrderingTest and ReceiveTest. Moreover, two UI's have been created for this feature: OrderHistoryUI and OrderWindowUI. The order history displays all of the Orders created, inbounded and received. The order window is where users will fill in the parameters to create a new order.

### CODE REVIEW

File: src/main/java/generate_order_use_case/OrderGenerator.java [hunk 1]
```diff
@@ -1,36 +1,29 @@
 package generate_order_use_case;
-
```
Comment (chenbriann) [inline src/main/java/generate_order_use_case/OrderGenerator.java:2]: Is it possible to incorporate the factory design pattern here?

**Review by chenbriann — APPROVED** [review]
Ordering feature looks good. Major improvements have been made to follow clean architecture and SOLID principles. Splitting ordering and receiving into use cases is a good way to ensure dependencies between layers are correct. I noticed that there is an order generator for making new orders, is it possible to incorporate the factory design pattern here? Also, I would consider making the ordering and receiving tests cover more, such as ordering new items and ordering items that already exist in the inventory.

**Review by PeterDushniku — APPROVED** [review]
Good work Monica,

Good job on adding Javadocs for your methods. They help clarify the usages of your methods a lot.
Also, your UI looks quite good. I like how the template for OrderWindowUI follows the layout for InventoryUI, it makes it easy to follow and consistent.
And, good work on clean architecture. I like how you made a use case for generate order in order to fit with the clean architecture principles.

**Review by guzzim2022 — APPROVED** [review]
This code, I believe, is an exemplary example of how to implement inventory ordering and receiving features. The implementation is very user-friendly; it constructs a screen to display all the orders and the history can be updated. Java Docs are clearly written and facilitate the reading of the code. There is some evidence of error management, for example checking if the order has not been received yet and only, in that case, receiving the order. I wonder if the data structure for the orders (currently a 2D string array) could be implemented using some more efficient data structure. Excellent work respecting the boundaries outlined in the clean architecture model and the SOLID design principle.

**Pull request merged**